File: simulator/src/main/java/it/unibo/scafi/simulation/gui/model/Node.java
package it.unibo.scafi.simulation.gui.model;

import java.awt.geom.Point2D;
import java.util.Map;
import java.util.Set;

/**
 * Created by chiara on 14/11/16.
 */
public interface Node {

    int getId();

    Object getExport();
    void setExport(Object export);

    Point2D getPosition();
    void setPosition(Point2D position);

    Set<Node> getNeighbours();
    boolean addNeighbour(Node neighbour);
    boolean addAllNeighbours(Set<Node> neighbours);
    boolean removeNeighbour(Node neighbour);
    void removeAllNeghbours();

    Set<Sensor> getSensors();
    Object getSensorValue(Sensor sensor);
    Object getSensorValue(String sensor);
    Map<Sensor, Object> getAllSensors();
    void setSensor(Sensor sensor, Object value);
    void setSensor(String sensor, Object value);

    void doAction(Action action);
}

File: simulator-gui/src/main/scala/it/unibo/scafi/simulation/gui/model/Node.scala
package it.unibo.scafi.simulation.gui.model

import java.awt.geom.Point2D

/**
  * Created by Varini on 14/11/16.
  * Converted/refactored to Scala by Casadei on 3/02/17
  */
trait Node {
  def id: Int

  var export: Any

  var position: Point2D

  def neighbours: Set[Node]

  def addNeighbour(neighbour: Node): Boolean

  def addAllNeighbours(neighbours: Set[Node]): Boolean

  def removeNeighbour(neighbour: Node): Boolean

  def removeAllNeghbours()

  def sensors: Map[Sensor, Any]

  def getSensorValue(sensorName: String): Any

  def setSensor(sensor: Sensor, value: Any)

  def setSensor(sensor: String, value: Any)
}